File: concepts/Refactoring/Java/Java Refactoring Best Practices/Bloaters/4Data-clumps/BaseProject/src/test/java/test/CheckoutHandlerTest.java
package test;


import com.onlineshop.CheckoutHandler;
import com.onlineshop.entities.Customer;
import com.onlineshop.entities.Order;
import com.onlineshop.items.Cheese;
import com.onlineshop.items.Chocolate;
import com.onlineshop.items.Item;
import com.sun.org.apache.xpath.internal.operations.Or;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.util.Arrays;
import java.util.List;


public class CheckoutHandlerTest {

    static List<Item> shoppingList = Arrays.asList(new Chocolate(), new Chocolate(), new Cheese());
    static CheckoutHandler checkout = new CheckoutHandler();



    @Test(description = "All is good, free delivery with gold membership")
    public void calculateTotalValidVoucherGoldMembership(){
        Customer customer1 = new Customer("GOLD", "MyStreet 123, US");

        double total = checkout.calculateTotal(new Order(shoppingList, "GIMME_DISCOUNT"), customer1);
        Assert.assertEquals(total, 4.27);
    }

    @Test(description = "invalid voucher")
    public void calculateTotalInValidVoucherGoldMembership(){
        Customer customer1 = new Customer("GOLD", "MyStreet 123, US");

        double total = checkout.calculateTotal(new Order(shoppingList, "DummyVoucher"), customer1);
        Assert.assertEquals(total, 4.5);

    }

    @Test(description = "invalid voucher, non-gold membership incurs US delivery fee")
    public void calculateTotalInValidVoucherNonGoldMembership(){
        Customer customer1 = new Customer("SILVER", "MyStreet 123, US");

        double total = checkout.calculateTotal(new Order(shoppingList, "DummyVoucher"), customer1);
        Assert.assertEquals(total, 9.5);
    }

    @Test(description = "invalid voucher, non-gold membership incurs Global delivery fee")
    public void calculateTotalInValidVoucherNonGoldMembershipNonUs(){
        Customer customer1 = new Customer("SILVER", "MyStreet 123, France");

        double total = checkout.calculateTotal(new Order(shoppingList, "DummyVoucher"), customer1);
        Assert.assertEquals(total, 14.5);
    }

}

File: software-engineering-concepts/Refactoring/PHP/PHP Refactoring Best Practices/1Bloaters/0BaseProject/tests/CheckoutHandlerTest.php
<?php

use app\handlers\CheckoutHandler;
use app\items\Cheese;
use app\items\Chocolate;
use PHPUnit\Framework\TestCase;

class CheckoutHandlerTest extends TestCase
{

    private $checkout;
    private $shoppingList;

    protected function setup(): void
    {
        $this->shoppingList = array(new Chocolate(), new Chocolate(), new Cheese());
        $this->checkout = new CheckoutHandler();
    }

    // "All is good, free delivery with gold membership")
    public function testCalculateTotalValidVoucherGoldMembership()
    {

        $total = $this->checkout->calculateTotal($this->shoppingList, "GIMME_DISCOUNT", "GOLD", "MyStreet 123, US");
        $this->assertEquals($total, 4.275);
    }

    // "invalid voucher")
    public function testCalculateTotalInValidVoucherGoldMembership()
    {

        $total = $this->checkout->calculateTotal($this->shoppingList, "DummyVoucher", "GOLD", "MyStreet 123, US");
        $this->assertEquals($total, 4.5);
    }

    // "invalid voucher, non-gold membership incurs US delivery fee")
    public function testCalculateTotalInValidVoucherNonGoldMembership()
    {
        $total = $this->checkout->calculateTotal($this->shoppingList, "DummyVoucher", "SILVER", "MyStreet 123, US");
        $this->assertEquals($total, 9.5);
    }

    //"invalid voucher, non-gold membership incurs Global delivery fee")
    public function testCalculateTotalInValidVoucherNonGoldMembershipNonUs()
    {
        $total = $this->checkout->calculateTotal($this->shoppingList, "DummyVoucher", "SILVER", "MyStreet 123, France");
        $this->assertEquals($total, 14.5);
    }

    protected function tearDown(): void
    {
        $this->checkout = null;
        $this->shoppingList = null;
    }
}
